pending-documentation.md

```diff
@@ -197,6 +197,50 @@ This will rewrite I to . if and only if the state cell contains I + 1.
 
 Note that in the case of Set and Map, one guarantee is that K1, K2, K3, and K4 represent /distinct/ elements. Pattern matching fails if the correct number of distinct elements cannot be found.
 
+## Set Variables
+
+### Motivation
+
+Set variables were introduced as part of Matching Mu Logic, the mathematical foundations for K.
+In Matching Mu Logic, terms evaluate to sets of values. This is useful for both capturing partiality (as in `3/0`) and
+capturing non-determinism (as in `3 #Or 5`).
+Consequently, symbol interpretation is extended to have a collective interpretation over sets of input values.
+
+Usually, K rules are given using regular variables, which expect that the term they match is both defined and has a
+unique interpretation.
+
+However, it is sometimes useful to have simplification rules which work over any kind of pattern, be it undefined or
+non-deterministic.  This behavior can be achieved by using set variables to stand for any kind of pattern.
+
+### Syntax
+
+Any variable prefixed by `@` will be considered a set variable.
+
+### Example
+
+Below is a simplification rule which motivated this extension:
+
+```
+  rule #Ceil(@I1:Int /Int @I2:Int) =>
+    {(@I2 =/=Int 0) #Equals true} #And #Ceil(@I1) #And #Ceil(@I2) 
+    [anywhere]
+```
+
+This rule basically says that `@I1:Int /Int @I2:Int` is defined if `@I1` and `@I2` are defined and
+`@I2` is not 0. using sets variables here is important as it allows the simplification rule to apply
+_any_ symbolic patterns, without caring whether they are defined or not. This allows simplifying
+the expression `#Ceil((A:Int /Int B:Int) / C:Int)` to
+`{(C =/=Int 0) #Equals true} #And #Ceil(C) #And ({(B =/=Int 0) #Equals true} #And #Ceil(B) #And #Ceil(A)` 
+
+See [kframework/kore#729](https://github.com/kframework/kore/issues/729) for more details.
+
+
```

Comment:
@xc93 can you provide a small and simple example where set variables match actual non-functional patterns?  Maybe the (Lin) axiom that defines LTL?  Or something about maps or SL?